src/Features/CSharp/Portable/MakeMethodSynchronous/CSharpMakeMethodSynchronousCodeFixProvider.cs

```diff
@@ -36,47 +39,66 @@ protected override SyntaxNode RemoveAsyncTokenAndFixReturnType(IMethodSymbol met
 
         private SyntaxNode FixMethod(IMethodSymbol methodSymbol, MethodDeclarationSyntax method, ITypeSymbol taskType, ITypeSymbol taskOfTType)
         {
-            var newReturnType = method.ReturnType;
+            var newReturnType = FixMethodReturnType(methodSymbol, method.ReturnType, taskType, taskOfTType);
+            SyntaxTokenList newModifiers = FixMethodModifiers(method.Modifiers, ref newReturnType);
+            return method.WithReturnType(newReturnType).WithModifiers(newModifiers);
+        }
+
+        private SyntaxNode FixLocalFunction(IMethodSymbol methodSymbol, LocalFunctionStatementSyntax localFunction, ITypeSymbol taskType, ITypeSymbol taskOfTType)
+        {
+            var newReturnType = FixMethodReturnType(methodSymbol, localFunction.ReturnType, taskType, taskOfTType);
+            SyntaxTokenList newModifiers = FixMethodModifiers(localFunction.Modifiers, ref newReturnType);
```

Comment:
how about taking in the return type, and returning it as a tuple with the modifiers.  Not a huge fan of 'ref' unless necessary for perf.